src/features/wallet/Wallet.tsx

```diff
@@ -67,7 +67,13 @@ const Wallet: FC<Props> = () => {
     signer,
     network
   );
-  // const previousVotingAddress = usePrevious(votingAddress);
+
+  const signingPK =
+    hotAddress && signingKeys[hotAddress]
+      ? signingKeys[hotAddress].privateKey
+      : votingAddress && signingKeys[votingAddress]
+      ? signingKeys[votingAddress].privateKey
+      : "";
```

Comment:
i feel like this could be simplified somewhat:
`const signingPK = signingKeys[hotAddress]?.privateKey || signingKeys[votingAddress]?.privateKey || ''`
its ok to index into `signingKeys` with a undefined value, it should return undefined since you never set that, and you can use the optional chaining to prevent runtime error indexing into undefined

Reply:
Null can't be used as a key, sadly. I could try to refactor so hotAddress is not string | null, though.

Reply:
ok i had a feeling that might have been the case. i think its fine as is then, dont bother changing